iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/iothub/serviceclient/RegistryManagerTests.java

```diff
@@ -657,4 +674,28 @@ private static boolean configWasDeletedSuccessfully(RegistryManager registryMana
         // configuration could still be retrieved, so it was not deleted successfully
         return false;
     }
+
+    private static RegistryManager buildRegistryManagerWithAzureSasCredential()
+    {
+        IotHubConnectionString iotHubConnectionStringObj = IotHubConnectionStringBuilder.createIotHubConnectionString(iotHubConnectionString);
+        IotHubServiceSasToken serviceSasToken = new IotHubServiceSasToken(iotHubConnectionStringObj);
+        AzureSasCredential azureSasCredential = new AzureSasCredential(serviceSasToken.toString());
+        RegistryManagerOptions options = RegistryManagerOptions.builder().httpReadTimeout(HTTP_READ_TIMEOUT).build();
+        return new RegistryManager(iotHubConnectionStringObj.getHostName(), azureSasCredential, options);
+    }
+
+    private static RegistryManager buildRegistryManagerWithTokenCredential()
+    {
+        IotHubConnectionString iotHubConnectionStringObj = IotHubConnectionStringBuilder.createIotHubConnectionString(iotHubConnectionString);
```

Comment:
Instead of constructing credentials per client type, can that not be extracted to a helper?

Reply:
I can extract the TokenCredential construction, sure